### PR TITLE
Feat (activation_calibration): speed-up by skipping quantization

### PR DESCRIPTION
**Pull request by Giuseppe5**

### CODE REVIEW

File: src/brevitas/core/quant/float.py [hunk 1]
```diff
@@ -86,10 +85,15 @@ def dequantize(self, y, scale):
...
+        scale = self.scaling_impl(x)
+        if self.observer_only:
+            y = x
+            saturating, inf_values, nan_values = self.float_clamp_impl.saturating, self.float_clamp_impl.inf_values, self.float_clamp_impl.nan_values
+        else:
+            y, scale = self.quantize(x, scale)
+            # after quantizing, clamp to special cases like NaN/inf if they are set
+            y, saturating, inf_values, nan_values = self.float_clamp_impl(
+                y, self.exponent_bit_width(), self.mantissa_bit_width(), self.exponent_bias())
         y = self.dequantize(y, scale)
```
Comment (Giuseppe5) [inline src/brevitas/core/quant/float.py:97]: Move one level in

File: src/brevitas/function/ops.py [hunk 2]
```diff
@@ -189,6 +189,10 @@ def min_int(signed: bool, narrow_range: bool, bit_width: Tensor) -> Tensor:
     return value
 
 
+def max_mantissa_func(val):
```
Comment (Giuseppe5) [inline src/brevitas/function/ops.py:192]: remove

**Comment by nickfraser** [comment]
How much speedup does this give for the examples you have tried?

**Review by nickfraser — APPROVED** [review]
LGTM - Thank you for the extra tests!

**Comment by Giuseppe5** [comment]
SDXL with minifloat quantization on 500 samples goes from 8h to 2h

**Pull request merged**